### ISSUE TITLE
Question about updating

### ISSUE THREAD

**Issue by mcasto** [aerotoad/neboa#9]
I just found neboa. At first glance, I **love** it. But I do have a question.

In the documentation, there's this example:

```
import { neboa } from 'neboa'

const db = neboa('database.db')

const Users = db.collection('users')

const user = Users.insert({ name: 'John Doe', age: 42 })

Users.update(user._id, { age: 43 })
```

That implies to me that the update function should result in a record like { name: 'John Doe', age: 43 }

But it doesn't. It results in { age: 43 }

Am I missing something, or is the documentation wrong (or outdated)?

To resolve this, I'll use ```Users.update(user._id, { ...user, age:43 }```

That works fine and it's easy, but I just want to make sure I'm not missing something.

**Comment by aerotoad** [comment]
@mcasto I think it's outdated docs from the first versions I wrote, Collection.update expects the latest version of the object to be passed and doesn't merge it for you.
It originally did the merge of the new data by itself but this could cause unexpected type issues when changing the model, so I decided to make it a userspace responsibility.

TL;DR: Pass the entire object you want to save, not just the properties you want to update.

**Comment by mcasto** [comment]
Got it. Thanks. I figured that was the case.

Follow up question. 

When I try to use neboa within the Electron app I'm building, I get the following error:

```
Error occurred in handler for 'get-shows': Error: The module '/Users/mikecasto/electron-projects/stage-sync/node_modules/better-sqlite3/build/Release/better_sqlite3.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 115. This version of Node.js requires
NODE_MODULE_VERSION 130.
```

After some googling, I found others resolved this by running electron-rebuild.

I attempted that but got the following:

```
An unhandled error occurred inside electron-rebuild
node-gyp failed to rebuild '/Users/mikecasto/electron-projects/stage-sync/node_modules/better-sqlite3'

Error: node-gyp failed to rebuild '/Users/mikecasto/electron-projects/stage-sync/node_modules/better-sqlite3'
    at ChildProcess.<anonymous> (/Users/mikecasto/electron-projects/stage-sync/node_modules/@electron/rebuild/lib/module-type/node-gyp/node-gyp.js:121:24)
    at ChildProcess.emit (node:events:518:28)
    at ChildProcess._handle.onexit (node:internal/child_process:293:12)
```

I've encountered this before when trying to use better-sqlite3 with Electron. I've never managed to resolve it and always just used sqlite3.

Do you have any tips on how to resolve the issue? Or is it possible for me to configure neboa to use sqlite3 instead of better-sqlite3?

**Comment by aerotoad** [comment]
This seems to be an issue with upstream electron and/or sqlite3. The couple of times it happened to me it was either a faulty python configuration or I needed to add better-sqlite3 as an external dependency to electron. Either way, if you don't manage to solve the issue, I'd recommend opening an issue with better-sqlite3 and/or electron since neboa is just a layer on top running fully in the js thread.

**Comment by mcasto** [comment]
OK. Thanks.
